instrumentor/controllers/instrumentationdevice/pods_webhook.go

```diff
@@ -34,6 +37,14 @@ type PodsWebhook struct {
 var _ webhook.CustomDefaulter = &PodsWebhook{}
 
 func (p *PodsWebhook) Default(ctx context.Context, obj runtime.Object) error {
+	logger := log.FromContext(ctx)
+
+	// In certain scenarios, the raw request can be utilized to retrieve missing details, like the namespace.
+	req, err := admission.RequestFromContext(ctx)
```

Comment:
Maybe worth mentioning this is relevant to Kubernetes <1.24 so we will know to delete this code once we no longer need to support version that old